manim/mobject/coordinate_systems.py

```diff
@@ -731,3 +739,298 @@ def get_coordinate_labels(self, *numbers, **kwargs):
+                "TAU radians": (1 / 20),
+                "degrees": (1 / 36),
+                "gradians": (1 / 40),
+                None: 1,
+            }[self.azimuth_units]
+            if azimuth_line_frequency is None
+            else azimuth_line_frequency
+        )
+        super().__init__(
+            color=color,
+            x_line_frequency=self.azimuth_frequency * TAU,
+            y_line_frequency=radius_line_frequency,
+            x_axis_config=radius_config,
+            x_min=None if (radius_max is None) else -radius_max,
+            x_max=radius_max,
+            y_min=-config["frame_x_radius"] if (radius_max is None) else -radius_max,
+            y_max=config["frame_x_radius"] if (radius_max is None) else radius_max,
+            **kwargs,
+        )
+        self.azimuth_offset = azimuth_offset
+        self.azimuth_direction = azimuth_direction
+        self.azimuth_buff = azimuth_buff
+        self.azimuth_scale = azimuth_scale
+        self.prepare_for_nonlinear_transform()
+        self.background_lines.apply_function(
+            lambda p: np.array([p[0] * np.sin(p[1]), p[0] * np.cos(p[1]), 0])
+        )
+
+    def prepare_for_nonlinear_transform(self, num_inserted_curves=50):
+        for mob in self.background_lines.family_members_with_points():
+            num_curves = mob.get_num_curves()
+            if num_inserted_curves > num_curves:
+                mob.insert_n_curves(num_inserted_curves - num_curves)
+        return self
+
+    def polar_to_point(self, radius, azimuth):
+        r"""Gets a point from polar coordinates.
+
+        Parameters
+        ----------
+        radius : :class:`float`
+            The coordinate radius (:math:`r`).
+
+        azimuth : :class:`float`
+            The coordinate azimuth (:math:`\theta`).
+
+        Returns
+        -------
+        :class:`numpy.ndarray`
+            The point.
+        """
+        return self.coords_to_point(radius * np.cos(azimuth), radius * np.sin(azimuth))
+
+    def pr2pt(self, radius, azimuth):
+        """Abbreviation for ``polar_to_point``"""
+        return self.polar_to_point(radius, azimuth)
+
+    def point_to_polar(self, point):
+        r"""Gets polar coordinates from a point.
+
+        Parameters
+        ----------
+        point : :class:`numpy.ndarray`
+            The point.
+
+        Returns
+        -------
+        Tuple[:class:`float`, :class:`float`]
+            The coordinate radius (:math:`r`) and the coordinate azimuth (:math:`\theta`).
+        """
+        x, y = self.point_to_coords(point)
+        return np.sqrt(x ** 2 + y ** 2), np.arctan2(y, x)
+
+    def pt2pr(self, point):
+        """Abbreviation for ``point_to_polar``"""
+        return self.point_to_polar(point)
+
+    def get_coordinate_labels(self, r_vals, a_vals, **kwargs):
+        if r_vals is None:
+            r_vals = [
+                r for r in self.get_x_axis().default_numbers_to_display() if r >= 0
```

Comment:
This will have to be adapted, the current documentation build fails (https://readthedocs.org/api/v2/build/13651256.txt) with:

```
  File "<string>", line 9, in <module>
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1325/lib/python3.8/site-packages/manim/mobject/coordinate_systems.py", line 1009, in add_coordinates
    self.add(self.get_coordinate_labels(r_vals, a_vals))
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1325/lib/python3.8/site-packages/manim/mobject/coordinate_systems.py", line 918, in get_coordinate_labels
    r for r in self.get_x_axis().default_numbers_to_display() if r >= 0
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/1325/lib/python3.8/site-packages/manim/mobject/mobject.py", line 530, in __getattr__
    raise AttributeError(f"{type(self).__name__} object has no attribute '{attr}'")
AttributeError: NumberLine object has no attribute 'default_numbers_to_display'
```

Reply:
Some of the function names have changed since the upstream merge. I've fixed that problem by changing it to `get_tick_range()`, but I'm getting an error out of `get_number_mobjects()` it seems `VGroup` won't allow `DecimalNumber`:

```
TypeError: All submobjects must be of type VMobject
```

Is this a bug?

Reply:
Try replacing `get_number_mobjects` with `add_numbers`, that's how I got it to work on my end.

Reply:
Yeah, that seems to work, still got a some strange rendering errors on my end though. Anyway, I'm on BST so I'll be off to bed soon. Thanks for your help @hydrobeam, once you get it working with #1161 you could start a PR on my branch and I could merge it tomorrow morning, would that be best?

Reply:
Sure sounds good! I'm also encountering some weird errors and I suspect it has to do with the code for  `NumberPlane` itself. I'll try to isolate the error.

Reply:
Any updates here?

Reply:
The issue turned out to be the new way `NumberPlane` was drawn (the box look). It intrduces odd errors where lines are off significant if the numbers aren't right. This was a bit unforseen, so I suggest we revert to the previous method of drawing `NumberPlane`

Also, since labels are added based on range, this means that there's a bit of an overflow in the x_direction of coordinate labels, especially for degrees (360 labels). We can either remove the label messages, or reintroduce `x_line_frequency`. I'm in favour of the first option.

Reply:
> The issue turned out to be the new way `NumberPlane` was drawn (the box look). It intrduces odd errors where lines are off significant if the numbers aren't right. This was a bit unforseen, so I suggest we revert to the previous method of drawing `NumberPlane`
> 

Sounds good to me. Could you take care of it (ideally within the next few hours)? We could bundle these changes to the PR I just opened, #1430.

> Also, since labels are added based on range, this means that there's a bit of an overflow in the x_direction of coordinate labels, especially for degrees (360 labels). We can either remove the labels along the radius, or reintroduce `x_line_frequency`. I'm in favour of the first option.

I'm not sure I understand. I would like to have the labels around the outer circle; for 360 degrees not every degree should be labeled.

Reply:
> Sounds good to me. Could you take care of it (ideally within the next few hours)? We could bundle these changes to the PR I just opened, #1430.

👍

> I'm not sure I understand. I would like to have the labels around the outer circle; for 360 degrees not every degree should be labeled.

My bad! I meant the labels across the radius in the inner circle that travel horizontally. The outer labels look as expected under the new change/